File: server/src/PersonalWebsite.Core/Models/Movie.cs
using PersonalWebsite.Core.Enums;

namespace PersonalWebsite.Core.Models;

public class Movie
{
    public string? Id { get; set; }
    public required string Name { get; set; }
    public required int Year { get; set; }
    public required Month Month { get; set; }
    public required int ReleaseYear { get; set; }
    public required Cinema Cinema { get; set; }
    public required bool IsNominated { get; set; }
    public required bool IsKino { get; set; }
    public required bool IsRetro { get; set; }
    public string? Alias { get; set; }
    public string? Zinger { get; set; }
    public string? ImageUrl { get; set; }
    public string? ImageObjectKey { get; set; }
    public string? AltImageUrl { get; set; }
    public string? AltImageObjectKey { get; set; }
}

File: client/src/types/Movie.ts
import Cinema from "./Cinema";

export default interface Movie {
  id: string;
  name: string;
  year: number;
  month: number;
  cinema: Cinema;
  releaseYear: number;
  zinger: string;
  isNominated: boolean;
  isKino: boolean;
  isRetro: boolean;
  alias: string;
  motif: string;
  imageUrl: string;
  imageObjectKey: string;
  altImageUrl: string;
  altImageObjectKey: string;
}
